### PR TITLE
Async data loading

### PR DESCRIPTION
**Pull request by rollaball**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Data can be fetched asynchronously data when a property `async` is set to `true`

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
Issue: #317

### CODE REVIEW

**Comment by rollaball** [comment]
@mathisonian 
is this good?
added 2 properties to data `async` and `initialValue`.
eg: `[data name:'myAsyncData' source:'url' async:true initialValue:'[]']`

**Comment by mathisonian** [comment]
Sweet that API looks good @rollaball 👍 

I haven't tested the changes locally yet, but the code updates so far look good. Is there any way to do it without adding the `axios` dependency?

**Comment by rollaball** [comment]
@mathisonian , I am not sure how we can do this without package such as axios. What do you recommend?

**Comment by mathisonian** [comment]
Fetch seems like a good solution 👍. This won't work in IE 11, so we should add a note to the docs about that. Users can include their own fetch polyfill if they need async data + IE11 support.

**Comment by mathisonian** [comment]
I'll do a full review of this today.

File: packages/idyll-document/src/utils/index.js [hunk 1]
```diff
@@ -238,12 +239,48 @@ export const filterIdyllProps = (props, filterInjected) => {
...
+
+      let dataPromise = new Promise(res => res(initialValue));
+
+      if (typeof fetch !== 'undefined') {
+        dataPromise = fetch(sourceValue)
+          .then(res => {
+            if (sourceValue.endsWith('.csv')) {
+              return res
+                .text()
+                .then(resString => parse(resString))
```
Comment (mathisonian) [inline packages/idyll-document/src/utils/index.js]: We should use the same options as the parsing that happens in the CLI so there aren't unexpected differences. There we call `parse(inputString, { cast: true, columns: true })`

File: packages/idyll-document/src/utils/index.js [hunk 2]
```diff
@@ -238,12 +239,48 @@ export const filterIdyllProps = (props, filterInjected) => {
...
+      ? getProperties(val).async.value
+      : false;
+    if (async) {
+      const initialValue = getProperties(val).initialValue
+        ? JSON.parse(getProperties(val).initialValue.value)
+        : [];
+
+      let dataPromise = new Promise(res => res(initialValue));
+
+      if (typeof fetch !== 'undefined') {
```
Comment (mathisonian) [inline packages/idyll-document/src/utils/index.js]: Can we print a warning if this check fails?

File: packages/idyll-document/src/utils/index.js (hunk 1 above)
Comment (rollaball) [reply packages/idyll-document/src/utils/index.js]: updated

File: packages/idyll-document/src/utils/index.js (hunk 2 above)
Comment (rollaball) [reply packages/idyll-document/src/utils/index.js]: updated

**Comment by rollaball** [comment]
@mathisonian , I have updated the code and added tests. The tests seem to be failing for some reason as it is unable to fetch the repo in sometimes so could you re-run it once.

**Comment by mathisonian** [comment]
Awesome work @rollaball ! 🎉

**Pull request merged**